api/src/pcapi/scripts/expire_pre_decree_credits/main.py

```diff
@@ -0,0 +1,89 @@
+"""
+https://github.com/pass-culture/pass-culture-main/blob/PC-35423-expire-deposits/api/src/pcapi/scripts/expire_pre_decree_credits/main.py
+"""
+
+from argparse import ArgumentParser
+import logging
+
+from dateutil.relativedelta import relativedelta
+from sqlalchemy.orm import Query
+from sqlalchemy.orm import joinedload
+
+from pcapi.app import app
+from pcapi.core.external.attributes.api import update_external_user
+from pcapi.core.finance.models import Deposit
+from pcapi.core.finance.models import DepositType
+from pcapi.core.history.models import ActionHistory
+from pcapi.core.history.models import ActionType
+from pcapi.core.users.models import User
+from pcapi.core.users.repository import find_user_by_email
+from pcapi.models import db
+from pcapi.settings import CREDIT_V3_DECREE_DATETIME
+
+
+logger = logging.getLogger(__name__)
+
+
+def expire_deposits(not_dry: bool, last_deposit_id: int = 0) -> None:
+    author = find_user_by_email("dan.nguyen@passculture.app")
+    if not author:
+        raise ValueError("author account not found")
+
+    deposits_query = _get_deposits_to_expire(last_deposit_id)
+    for deposit in deposits_query.yield_per(500):
+        _expire_deposit(deposit, author)
+
+        if not_dry:
+            db.session.flush()
+            update_external_user(deposit.user)
+
+
+def _get_deposits_to_expire(last_deposit_id: int) -> Query:
+    two_years_before_decree = CREDIT_V3_DECREE_DATETIME - relativedelta(years=2)
+    return (
+        Deposit.query.filter(
+            Deposit.type == DepositType.GRANT_18,
+            Deposit.dateCreated < two_years_before_decree,
+            Deposit.expirationDate >= CREDIT_V3_DECREE_DATETIME,
+            Deposit.id > last_deposit_id,
+        )
+        .order_by(Deposit.id)
+        .options(joinedload(Deposit.user))
+    )
+
+
+def _expire_deposit(deposit: Deposit, author: User) -> None:
+    logger.info(
+        "expiring deposit %s, that was created the %s",
+        deposit.id,
+        deposit.dateCreated,
+        extra={"user_id": deposit.user.id},
+    )
+
+    deposit.expirationDate = deposit.dateCreated + relativedelta(years=2)
+
+    action_history = ActionHistory(
+        actionType=ActionType.FRAUD_INFO_MODIFIED,
```

Comment:
`FRAUD_INFO_MODIFIED` correspond aux commentaires qui ne sont visibles que par l'équipe fraude. Ici, je pense qu'il est préférable que le commentaire soit visible par tous, dont le support, afin de comprendre l'enchaînement si besoin. Comme il n'y a pas de extraData.modified_info, je propose tout simplement `COMMENT` : 
```suggestion
        actionType=ActionType.COMMENT,
```